### ISSUE TITLE
model.eval() in train.py

### ISSUE THREAD

**Issue by githwd2016** [shubhamagarwal92/mmd#14]
In `train.py` method `evaluate`, I find you use `model.eval()` to disable `dropout`. However, after `evaluate`, I didn't find `model.train()` to enable `dropout`. It seems that once `evaluate` called, `dropout` will be disabled. Does this influence final performance?

**Comment by Shandilya21** [comment]
@githwd2016 , I am unsure about the final performance but I was facing an issues while training.  It sets model to eval when generating samples and logging, but never sets it back to train after returning to training and got error like  [link](https://github.com/pytorch/pytorch/issues/7961).
 
Solution 
after evaluate function in  `train.py` line 132, add model.train( )

**Comment by shubhamagarwal92** [comment]
Hmm interesting observation! 

@Shandilya21 If you are facing an error, that might be because of different pytorch version. It should run on the version mentioned. (I guess they have also removed/changed these flags from 0.4.1) 

It would be interesting to see if there was an actual performance difference. Do keep me updated. Thanks.

**Comment by Shandilya21** [comment]
Hi @shubhamagarwal92 , 
I even tried with pytorch 0.3.1 version but it shows same problem while training. Initially, I have pytorch 0.4.1 version, i switched down to 0.3.1, but replicate the same problem. however, i put model.train() after the evaluation method in   `train.py`,  it works fine for me.

yeah do let you know regarding the actual performance difference

**Comment by githwd2016** [comment]
Another bug: in `train_and_translate.sh`, `ANNOY_PKL` is set as `ImageUrlToIndex.pkl`, while in `train.py`, the image is saved as `FileName`, which could result in that all image representation is **zero tensor**. Use `FileNameMapToIndex.pkl` rather than `ImageUrlToIndex.pkl`.

**Comment by Shandilya21** [comment]
Hi, @githwd2016 
I think `ImageUrlToIndex.pkl` is for HRED model_type not for MultimodalHRED_type. In HRED type model all the images representation are considered to be zero tensor for concatenation. whereas in MultimodalHRED, you may use `FileNameMapToIndex.pkl` rather than `ImageUrlToIndex.pkl`.

**Comment by shubhamagarwal92** [comment]
I asked the original authors what `.pkl` to use and didnt get any reply. I guess I finally ended up trying with both and probably didn't get any much difference. Probably that is why, I have been using them interchangably.

**Comment by shubhamagarwal92** [comment]
As always, if you see any bug, do not hesitate to raise a PR. Thanks.

**Comment by shubhamagarwal92** [comment]
Closing this now.
